File: tablet/Resources/menu/AnonymousMenu.js
// AffectiveMenu.js

/*
    Copyright (C) 2012-2015 Rudolf Cardinal (rudolf@pobox.com).
    Department of Psychiatry, University of Cambridge.
    Funded by the Wellcome Trust.

    This file is part of CamCOPS.

    Licensed under the Apache License, Version 2.0 (the "License");
    you may not use this file except in compliance with the License.
    You may obtain a copy of the License at

        http://www.apache.org/licenses/LICENSE-2.0

    Unless required by applicable law or agreed to in writing, software
    distributed under the License is distributed on an "AS IS" BASIS,
    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
    See the License for the specific language governing permissions and
    limitations under the License.
*/

/*jslint node: true, newcap: true */
"use strict";
/*global L */

module.exports = function AnonymousMenu() {

    var MenuWindow = require('menulib/MenuWindow'),
        UICONSTANTS = require('common/UICONSTANTS'),
        ALLTASKS = require('common/ALLTASKS'),
        data = [
            ALLTASKS.TASKLIST.GMCPQ,
            ALLTASKS.TASKLIST.REFERRER_SATISFACTION_GEN,
        ],
        self = new MenuWindow({
            title: L('menutitle_anonymous'),
            icon: UICONSTANTS.ICON_CHOOSE_PATIENT,
            data: data,
        });

    return self;
};

File: tablet_qt/menu/anonymousmenu.cpp
/*
    Copyright (C) 2012, University of Cambridge, Department of Psychiatry.
    Created by Rudolf Cardinal (rnc1001@cam.ac.uk).

    This file is part of CamCOPS.

    CamCOPS is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.

    CamCOPS is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
    GNU General Public License for more details.

    You should have received a copy of the GNU General Public License
    along with CamCOPS. If not, see <https://www.gnu.org/licenses/>.
*/

#include "anonymousmenu.h"

#include "common/uiconst.h"
#include "lib/uifunc.h"
#include "menulib/menuitem.h"
#include "tasks/apeqpt.h"
#include "tasks/gmcpq.h"
#include "tasks/perinatalpoem.h"
#include "tasks/referrersatisfactiongen.h"

AnonymousMenu::AnonymousMenu(CamcopsApp& app) :
    MenuWindow(app, uifunc::iconFilename(uiconst::ICON_ANONYMOUS))
{
}

QString AnonymousMenu::title() const
{
    return tr("Anonymous questionnaires");
}

void AnonymousMenu::makeItems()
{
    m_items = {
        // Seems inappropriate: MAKE_CHANGE_PATIENT(m_app),
        MAKE_TASK_MENU_ITEM(Apeqpt::APEQPT_TABLENAME, m_app),
        MAKE_TASK_MENU_ITEM(GmcPq::GMCPQ_TABLENAME, m_app),
        MAKE_TASK_MENU_ITEM(PerinatalPoem::PERINATAL_POEM_TABLENAME, m_app),
        MAKE_TASK_MENU_ITEM(
            ReferrerSatisfactionGen::REF_SATIS_GEN_TABLENAME, m_app
        ),
    };
}
